src/main/java/com/vaadin/recipes/recipe/clientsideupload/FileUploadEndpoint.java

```diff
@@ -0,0 +1,19 @@
+package com.vaadin.recipes.recipe.clientsideupload;
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+
+@RestController
+public class FileUploadEndpoint {
+
+    @PostMapping("/api/fileupload")
+    public void handleFileUpload(@RequestParam("file") MultipartFile uploadedFile) throws IOException {
+        File file = new File(System.getProperty("user.dir") + "/uploadedFile");

```

Comment:
I'm not sure if this could cause any trouble on the server.

Reply:
You should probably just show a notification like "upload filename successful on a successful return. Anonymous users uploading files on a server doesn't sound good 😁

Reply:
That makes sense. Changed the recipe to show a notification. Thanks for the suggestion 🙏